server/src/main/java/org/elasticsearch/search/aggregations/AggregationPhase.java

```diff
@@ -25,36 +26,34 @@
  * Aggregation phase of a search request, used to collect aggregations
  */
 public class AggregationPhase {
-
     @Inject
     public AggregationPhase() {}
 
     public static void preProcess(SearchContext context) {
         if (context.aggregations() == null) {
             return;
         }
-        BucketCollector bucketCollector;
-        try {
-            context.aggregations().aggregators(context.aggregations().factories().createTopLevelAggregators());
-            bucketCollector = MultiBucketCollector.wrap(true, List.of(context.aggregations().aggregators()));
-            bucketCollector.preCollection();
-        } catch (IOException e) {
-            throw new AggregationInitializationException("Could not initialize aggregators", e);
-        }
+        AggregationCollectorManager collectorManager = new AggregationCollectorManager(context);
         if (context.aggregations().factories().context() != null
             && context.aggregations().factories().context().isInSortOrderExecutionRequired()) {
             TimeSeriesIndexSearcher searcher = new TimeSeriesIndexSearcher(context.searcher(), getCancellationChecks(context));
             try {
-                searcher.search(context.rewrittenQuery(), bucketCollector);
+                // TODO: we are executing here the time series aggregation and let we will execute the query?
```

Comment:
This is only for aggregations if the `time_series` aggregation is used. The collecting top docs for main query is still done in `QueryPhase`.  If the `time_series` aggregation is not used then both top docs and aggregations are computed in `QueryPhase`.